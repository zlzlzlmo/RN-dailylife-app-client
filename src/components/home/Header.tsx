import { StyleSheet, Text, TextInput, View } from "react-native";
import React, { useState } from "react";
import { colors } from "../../styles/variables/color";
import { AntDesign } from "@expo/vector-icons";

const Header = () => {
  const [searchValue, setSearchValue] = useState<string>("");

  const handleSearchValue = (text: string) => {
    setSearchValue(text);
  };

  const resetSearchValue = () => {
    setSearchValue("");
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputContainer}>
        <AntDesign
          style={styles.searchIcon}
          name="search1"
          size={24}
          color={colors.inputPlacehoderColor}
        />
        <TextInput
          style={styles.input}
          placeholder="게시글을 검색해보세요."
          onChangeText={handleSearchValue}
          value={searchValue}
        />
        {searchValue.length > 0 && (
          <AntDesign
            style={styles.resetIcon}
            name="closecircle"
            size={24}
            color={colors.inputPlacehoderColor}
            onPress={resetSearchValue}
          />
        )}
      </View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.whiteColor,
    padding: 20,
  },
  inputContainer: {
    position: "relative",
    backgroundColor: "blue",
    borderRadius: 10,
  },
  searchIcon: {
    position: "absolute",
    zIndex: 1,
    top: "50%",
    transform: [{ translateY: -10 }],
    left: 10,
  },
  resetIcon: {
    position: "absolute",
    top: "50%",
    transform: [{ translateY: -10 }],
    right: 10,
  },
  input: {
    backgroundColor: colors.inputBackgroundColor,
    height: 50,
    paddingHorizontal: 50,
    overflow: "hidden",
  },
});
