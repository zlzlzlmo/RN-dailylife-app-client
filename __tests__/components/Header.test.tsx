import { render, screen } from "@testing-library/react-native";
import Header from "../../src/components/home/Header";
import renderer from "react-test-renderer";

describe("<Header/>", () => {
  test("renders correctly", () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });
});
