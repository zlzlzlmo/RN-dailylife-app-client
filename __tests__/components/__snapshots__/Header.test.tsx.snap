// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "blue",
        "borderRadius": 10,
        "position": "relative",
      }
    }
  >
    <Text />
    <TextInput
      onChangeText={[Function]}
      placeholder="게시글을 검색해보세요."
      style={
        Object {
          "backgroundColor": "#F7FAFB",
          "height": 50,
          "overflow": "hidden",
          "paddingHorizontal": 50,
        }
      }
      value=""
    />
  </View>
</View>
`;
